go-controller/pkg/clustermanager/healthcheck/provider.go

```diff
@@ -0,0 +1,155 @@
+package healthcheck
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+
+	coreinformers "k8s.io/client-go/informers/core/v1"
+)
+
+type HealthState int
+
+const (
+	// UNKNOWN is the health state of any node that is not being monitored.
+	// Services should not make any determination based on this state and should
+	// consider the node potentially available for scheduling.
+	UNKNOWN HealthState = iota
+
+	// AVAILABLE is the health state of a node that is reachable and thought to
+	// have a healthy ovn-kube running. Services should consider the node
+	// available for scheduling.
+	AVAILABLE
+
+	// UNREACHABLE is the health state of a node that is not reachable. Services
+	// should not consider the node available for scheduling and should attempt
+	// to unschedule from it as well.
+	UNREACHABLE
+)
+
+func (s HealthState) String() string {
+	switch s {
+	case AVAILABLE:
+		return "AVAILABLE"
+	case UNREACHABLE:
+		return "UNREACHABLE"
+	}
+	return "UNKNOWN"
+}
+
+// Provider of node health state information
+type Provider interface {
+	// Register a consumer to be informed about node health state changes
+	Register(consumer Consumer)
+	// GetHealthState of the node
+	GetHealthState(node string) HealthState
+	// MonitorHealthState notifies interest of a consumer to monitor the health
+	// state of a node.
+	MonitorHealthState(node string)
+}
+
+// Consumer of node health state information
+type Consumer interface {
+	// Name the consumer goes by for logging purposes
+	Name() string
+	// HealthStateChanged informs consumers of node health state changes. A
```

Comment:
its not clear from this if the function informs consumers of the node health. It returns nothing. Also is this function triggered when a change happens, or is it left to the consumer to call it and it returns if something changed since it was last queried?

Reply:
The consumer interface is implemented by consumers of health state information. This method is called from the healthcheck controller onto the consumer to inform that the health state of a node has changed. The consumer is supposed to trigger its reconciliation loop, if any, where it will query the actual health state of the node and act accordingly.

Reply:
So is it accurate to say that HealthStateChanged is a callback that is triggered by the healthcheck controller when there is a state change? Can we update the comments to make that more obvious?